irods/test/user_group_test.py

```diff
@@ -88,6 +88,28 @@ def generator(p = OLDPASS):
             shutil.rmtree(ENV_DIR)
             ses.users.remove('alice')
 
+    def test_modifying_password_at_various_lengths__328(self):
+        ses = self.sess
+        try:
+            pw_lengths = [3, MAX_PASSWORD_LENGTH - 8, 26, 25, 24 ]
+            to_from = [('a'*x,'b'*y) for x in pw_lengths for y in pw_lengths ]
```

Comment:
Definitely need a comment explaining this. Or perhaps a better variable name?

The nested loops are a little hard to understand. Wait, are the loops nested at all?

It's fine if the answer is for me to learn a bit more Python too :-)

Reply:
They are nestet. `[ (a,b) for a in L1 for b in L1 ]` would be the same as the value of L2 at the end of
```
L2=[ ]
for a in L1:
  for b in L1:
    L2.append( (a,b) )
```
```

Reply:
I guess that is going to take me some time to get used to.

Perhaps a comment explaining what `to_from` represents is good enough.

Reply:
Python (for better or worse) is the only language I know that features list- and dict-comprehensions. maybe for good reasons...

Reply:
I think it's fine. More on me than anything else.

Reply:
Made this clearer now.